swaybar/tray/item.c

```diff
@@ -494,23 +494,33 @@ uint32_t render_sni(cairo_t *cairo, struct swaybar_output *output, double *x,
 	}
 
 	int padded_size = icon_size + 2*padding;
-	*x -= padded_size;
+	double padded_surface_size = (double) padded_size / output->scale;
+	double x_in_pixels = (*x * output->scale) - padded_size;
 	int y = floor((height - padded_size) / 2.0);
 
+	// reset scale
```

Comment:
Can we add a small comment here explaining why we're doing this? Something like "Icons need to be rendered in buffer-local space, otherwise they end up being blurry"